syntax = "proto3";

package chatroom;

// 对后台服务器：负载状态报告的报文
message StatusReportReq {
    uint32 server_id = 1;
    uint32 load = 2;
    // 以下两个字段状态服务器会忽略，但未来可能用上
    uint32 cpu_usage = 3;
    uint32 mem_usage = 4;
}

// 对网关：检查用户是否在线的报文
message UserCheckReq {
    string user_id = 1;
    bool kick = 2;  // 设为true时，用户在线的情况会被强制下线
}

// 对网关：请求最低负载的服务器的报文
message MinimalLoadServerReq {};

message ServerAddrResp {
    uint32 ret = 1;
    uint32 server_id = 2;
    string server_addr = 3;
};

// RPC调用的通用响应报文
message GeneralResp {
    uint32 ret = 1;
}

// 状态服务器的rpc接口定义
service StatusService {
    rpc ReportServerLoad (StatusReportReq) returns (GeneralResp) {}
    rpc CheckUserOnline (UserCheckReq) returns (GeneralResp) {}
    rpc CheckMinimalLoadServer (MinimalLoadServerReq) returns (ServerAddrResp) {}
}